sentry_streams/src/run.rs

```diff
@@ -0,0 +1,99 @@
+use std::ffi::OsString;
+
+use clap::Parser;
+use pyo3::prelude::*;
+
+use crate::utils::traced_with_gil;
+
+#[derive(Parser, Debug)]
+pub struct Args {
+    #[command(flatten)]
+    pub py_config: PyConfig,
+
+    #[command(flatten)]
+    pub runtime_config: RuntimeConfig,
+}
+
+#[derive(Parser, Debug)]
+pub struct PyConfig {
+    /// Path of the python interpreter executable
+    #[arg(short, long)]
+    pub exec_path: Option<OsString>,
+
+    #[arg(short, long)]
+    /// Paths to the relevant python modules
+    pub module_paths: Option<Vec<OsString>>,
+}
+
+#[derive(Parser, Debug)]
+pub struct RuntimeConfig {
+    /// The name of the adapter
+    #[arg(short, long)]
+    pub adapter_name: String,
+
+    /// The deployment config file path. Each config file currently corresponds to a specific pipeline.
+    #[arg(short, long)]
+    pub config_file: OsString,
+
+    /// The segment id to run the pipeline for
+    #[arg(short, long)]
+    pub segment_id: Option<String>,
+
+    /// The name of the application
+    pub application_name: String,
+}
+
+pub fn run(args: Args) -> Result<(), Box<dyn std::error::Error>> {
+    pyo3::prepare_freethreaded_python();
```

Comment:
I don't think we can use freethreaded python now. Assuming this starts python with freethreading. That's because, as I understand it, it requires the application to take care of race conditions more and our application does not do that.